api/v1/db/oracledb/students-dao.js

```diff
@@ -8,24 +9,26 @@ const conn = appRoot.require('api/v1/db/oracledb/connection');
 /**
  * Return serialized resource(s) by unique ID
  *
- * @param {string} id The unique ID for resource(s)
+ * @param {string} osuId 9 digits OSU ID
  * @param {string} sql The SQL statement that is executed
  * @param {Function} serializer Resource serializer function
  * @param {boolean} isSingleton A Boolean value represents the resource should be singleton or not
+ * @param {object} binds Bind parameters
  * @param {object} params A key-value pair params object
  * @returns {Promise<object>} Promise object represents serialized resource(s)
  */
-const getResourceById = async (id, sql, serializer, isSingleton, params) => {
+const getResourceById = async (osuId, sql, serializer, isSingleton, binds, params) => {
```

Comment:
From our discussion, I think the logic for adding `osuId` and `term` to `binds` could be handled by this function but as you pointed out, the more complicated logic can remain in the other functions to avoid bloating this one